### PR TITLE
feat(query): add --input-data option

### LINKED ISSUE #3807: Inject custom data in queries
[Following @mukeshpilaniya suggestion](https://gitter.im/kics-io/community?at=6099011d20d4f02631b25f60), since Rego has the ability to user injects custom data through the store, KICS should allows the user to give a custom JSON to overwrite default data in a simple way.

### PR DESCRIPTION
**Pull request by felipe-avelar**
Closes #3807 

**Proposed Changes**
- Added a way to users input some data to customize queries with their own data.
This implementation will require to update queries that makes sense to customize received data, like passwords queries, for example.
To use this feature, we must add `data.<data_variable>` on query and create a file `data.json` that will have all data variables as keys.
To customize data, you need to use `--input-data` flag, pointing to a folder with files in the format <query_id>.json and replace ALL keys with your own values.

This will be documented as soon as this PR is merged to master

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/common/passwords_and_secrets_in_infrastructure_code/metadata.json [hunk 1]
```diff
@@ -5,5 +5,10 @@
   "category": "Secret Management",
   "descriptionText": "Query to find passwords and secrets in infrastructure code.",
   "descriptionUrl": "https://kics.io/",
-  "platform": "Common"
+  "platform": "Common",
+  "override": {
+    "override": {
+      "queryName": "teste"
```
Comment (joaoReigota1) [inline assets/queries/common/passwords_and_secrets_in_infrastructure_code/metadata.json:11]: @felipe-avelar Is this supposed to be like this, or was it for testing purposes?

**Review by joaoReigota1 — CHANGES_REQUESTED** [review]

File: assets/queries/common/passwords_and_secrets_in_infrastructure_code/data.json [hunk 2]
```diff
@@ -0,0 +1,148 @@
+{
+  "ar": [
```
Comment (rogeriopeixotocx) [inline assets/queries/common/passwords_and_secrets_in_infrastructure_code/data.json:2]: How about taking this opportunity to increase readability and rename this array to default passwords.
```suggestion
  "defaultPasswords": [
```

File: assets/queries/common/passwords_and_secrets_in_infrastructure_code/data.json [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    "11111",
+    "123abc",
+    "asdf",
+    "centos",
+    "888888",
+    "54321",
+    "password123",
+    "pa$$"
+  ],
+  "bl": [
```
Comment (rogeriopeixotocx) [inline assets/queries/common/passwords_and_secrets_in_infrastructure_code/data.json:104]:
```suggestion
  "blackList": [
```

File: docs/commands.md [hunk 4]
```diff
@@ -68,6 +68,7 @@ Flags:
                                      cannot be provided with query exclusion flags
                                      can be provided multiple times or as a comma separated string
                                      example: 'e69890e6-fce5-461d-98ad-cb98318dfc96,4728cd65-a20c-49da-8b31-9c08b423e4db'
+      --input-data string            path to input data files
```
Comment (rogeriopeixotocx) [inline docs/commands.md:71]:
```suggestion
      --input-data string            path to query input data files
```

File: docs/dockerhub.md [hunk 5]
```diff
@@ -96,6 +96,7 @@ Flags:
                                      cannot be provided with query exclusion flags
                                      can be provided multiple times or as a comma separated string
                                      example: 'e69890e6-fce5-461d-98ad-cb98318dfc96,4728cd65-a20c-49da-8b31-9c08b423e4db'
+      --input-data string            path to input data files
```
Comment (rogeriopeixotocx) [inline docs/dockerhub.md:99]:
```suggestion
      --input-data string            path to query input data files
```

File: e2e/fixtures/assets/scan_help [hunk 6]
```diff
@@ -28,6 +28,7 @@ Flags:
                                      cannot be provided with query exclusion flags
                                      can be provided multiple times or as a comma separated string
                                      example: 'e69890e6-fce5-461d-98ad-cb98318dfc96,4728cd65-a20c-49da-8b31-9c08b423e4db'
+      --input-data string            path to input data files
```
Comment (rogeriopeixotocx) [inline e2e/fixtures/assets/scan_help:31]:
```suggestion
      --input-data string            path to query input data files
```

File: internal/console/scan.go [hunk 7]
```diff
@@ -293,6 +295,7 @@ func setBoundFlags(flagName string, val interface{}, cmd *cobra.Command) {
 	}
 }
 
+//nolint
```
Comment (rogeriopeixotocx) [inline internal/console/scan.go:298]: complexity/function too long warning? what was it

File: pkg/engine/inspector.go [hunk 8]
```diff
@@ -133,11 +135,14 @@ func NewInspector(
 		case <-ctx.Done():
 			return nil, nil
 		default:
-			opaQuery, err := rego.New(
+			var opaQuery rego.PreparedEvalQuery
+			store := inmem.NewFromReader(bytes.NewBufferString(metadata.InputData))
```
Comment (rogeriopeixotocx) [inline pkg/engine/inspector.go:139]: nice

File: pkg/engine/source/filesystem.go [hunk 9]
```diff
@@ -279,3 +301,37 @@ func getPlatform(queryPath string) string {
...
+	customDataJSON := map[string]interface{}{}
+	if unmarshalError := json.Unmarshal([]byte(queryInputData), &dataJSON); unmarshalError != nil {
+		return "", errors.Wrapf(unmarshalError, "failed to merge query input data")
+	}
+	if unmarshalError := json.Unmarshal([]byte(customInputData), &customDataJSON); unmarshalError != nil {
+		return "", errors.Wrapf(unmarshalError, "failed to merge query input data")
+	}
+
+	for key, value := range customDataJSON {
+		dataJSON[key] = value
```
Comment (rogeriopeixotocx) [inline pkg/engine/source/filesystem.go:330]: so custom data always override default... ok!

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
looks good, just a few suggestions

File: internal/console/scan.go (hunk 7 above)
Comment (felipe-avelar) [reply internal/console/scan.go:298]: Exactly, the problem in this function is, every time we add a new flag, this function will grow. I couldn't think anyway to reduce the number of lines, do you have any suggestion?

**Review by rogeriopeixotocx — APPROVED** [review]

**Review by joaoReigota1 — APPROVED** [review]
LGTM

**Pull request merged**